### ISSUE TITLE
[bug] conan_server fails if search unknown package in revision_enabled mode

### ISSUE THREAD

**Issue by avpalienko** [conan-io/conan#13067]
### Environment details

* Operating System+version: Oracle Linux 8.7
* Compiler+version: Any
* Conan version: 1.58.0
* Python version: 3.6

### Steps to reproduce

 conan config set general.revisions_enabled=1
 conan search 11111mozilla/1.0.0@user/channel -r=ol87

### Logs

Client: 

> conan search 11111mozilla/1.0.0@user/channel -r=ol87
ERROR: HTTPConnectionPool(host='192.168.56.102', port=9300): Max retries exceeded with url: /v2/conans/11111mozilla/1.0.0/user/channel/search (Caused by ResponseError('too many 500 error responses')). [Remote: ol87]

Server console: 

>192.168.56.1 - - [07/Feb/2023 03:51:05] "GET /v1/ping HTTP/1.1" 200 0
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/conans/server/rest/bottle_plugins/return_handler.py", line 34, in wrapper
    return callback(*args, **kwargs)  # kwargs has :xxx variables from url
  File "/usr/local/lib/python3.6/site-packages/conans/server/rest/bottle_plugins/authorization_header.py", line 48, in wrapper
    return callback(*args, **kwargs)  # kwargs has :xxx variables from url
  File "/usr/local/lib/python3.6/site-packages/conans/server/rest/controller/v2/search.py", line 33, in search_packages
    info = search_service.search_packages(ref, query)
  File "/usr/local/lib/python3.6/site-packages/conans/server/service/common/search.py", line 84, in search_packages
    info = search_packages(self._server_store, reference, query, look_in_all_rrevs)
  File "/usr/local/lib/python3.6/site-packages/conans/server/service/common/search.py", line 64, in search_packages
    latest_rev = server_store.get_last_revision(ref).revision
AttributeError: 'NoneType' object has no attribute 'revision'
192.168.56.1 - - [07/Feb/2023 03:51:05] "GET /v2/conans/11111mozilla/1.0.0/user/channel/search HTTP/1.1" 500 45
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/conans/server/rest/bottle_plugins/return_handler.py", line 34, in wrapper
    return callback(*args, **kwargs)  # kwargs has :xxx variables from url
  File "/usr/local/lib/python3.6/site-packages/conans/server/rest/bottle_plugins/authorization_header.py", line 48, in wrapper
    return callback(*args, **kwargs)  # kwargs has :xxx variables from url
  File "/usr/local/lib/python3.6/site-packages/conans/server/rest/controller/v2/search.py", line 33, in search_packages
    info = search_service.search_packages(ref, query)
  File "/usr/local/lib/python3.6/site-packages/conans/server/service/common/search.py", line 84, in search_packages
    info = search_packages(self._server_store, reference, query, look_in_all_rrevs)
  File "/usr/local/lib/python3.6/site-packages/conans/server/service/common/search.py", line 64, in search_packages
    latest_rev = server_store.get_last_revision(ref).revision
AttributeError: 'NoneType' object has no attribute 'revision'
192.168.56.1 - - [07/Feb/2023 03:51:05] "GET /v2/conans/11111mozilla/1.0.0/user/channel/search HTTP/1.1" 500 45
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/conans/server/rest/bottle_plugins/return_handler.py", line 34, in wrapper
    return callback(*args, **kwargs)  # kwargs has :xxx variables from url
  File "/usr/local/lib/python3.6/site-packages/conans/server/rest/bottle_plugins/authorization_header.py", line 48, in wrapper
    return callback(*args, **kwargs)  # kwargs has :xxx variables from url
  File "/usr/local/lib/python3.6/site-packages/conans/server/rest/controller/v2/search.py", line 33, in search_packages
    info = search_service.search_packages(ref, query)
  File "/usr/local/lib/python3.6/site-packages/conans/server/service/common/search.py", line 84, in search_packages
    info = search_packages(self._server_store, reference, query, look_in_all_rrevs)
  File "/usr/local/lib/python3.6/site-packages/conans/server/service/common/search.py", line 64, in search_packages
    latest_rev = server_store.get_last_revision(ref).revision
AttributeError: 'NoneType' object has no attribute 'revision'
192.168.56.1 - - [07/Feb/2023 03:51:05] "GET /v2/conans/11111mozilla/1.0.0/user/channel/search HTTP/1.1" 500 45

**Comment by memsharded** [comment]
Hi @avpalienko 

Thanks for your report.
I have reproduced the error and proposed a fix in https://github.com/conan-io/conan/pull/13070. It shouldn't be an issue anymore in 2.0, but as it is related to revisions, I guess it might help with the migration.

I have also checked, and ArtifactoryCE (it is completely free) is not suffering from this issue, in case you want to try it before the next release is out.
